src/List.mo

```diff
@@ -1643,22 +1858,32 @@ module {
   ///
   /// *Runtime and space assumes that `compare` runs in O(1) time and space.
   public func compare<T>(list1 : List<T>, list2 : List<T>, compare : (T, T) -> Order.Order) : Order.Order {
-    let size1 = size(list1);
-    let size2 = size(list2);
-    let minSize = if (size1 < size2) { size1 } else { size2 };
+    let blocks1 = list1.blocks;
+    let blocks2 = list2.blocks;
+    let blockCount = Nat.min(blocks1.size(), blocks2.size());
```

Comment:
Is this part worth it? The size calculation only happens once for the whole list. Cost does not depend on length of inputs.